docs/architecture/deep-dives/authentication/index.md

```diff
@@ -0,0 +1,325 @@
+request a new `access_token` without prompting the user to authenticate again.
+
+The claims on the `access_token` are used to initialize the user's `AccountProfile` in state, and
+the `access_token` and `refresh_token` are stored on the `AccountTokens` in state for use on
+subsequent API requests.
+
+### Use the Decryption Information to Prepare for Decryption
+
+The `AccountKeys` (stored in `account.keys` in state) are also set up in the user's account using
+the information provided in the token response.
+
+The important keys that are set at this point are:
+
+#### Device Key (`deviceKey`)
+
+The Device Key is the symmetric encryption key stored on the user's trusted device and is used to
+decrypt the user's symmetric encryption key when using Trusted Device Encryption.
+
+It is not returned on the request (as it never leaves the device), but it is set on the
+authenticated user's account state at this point.
+
+#### User Key (`userKey`)
+
+The User Key is the symmetric encryption key used to decrypt the user's vault.
+
+The specifics of how the User Key is set are defined in each login strategy:
+
+- **Password and Login with Device strategies**
+  - The `Key` in the token response is stored in `userKeyMasterKey` in the account state, and the
+    `masterKey` already in state from the user entering their master password is used to decrypt and
+    store the `userKey`.
+- **API strategy**
+  - The `Key` in the token response is stored in `userKeyMasterKey` in the account state.
+  - If the `ApiUseKeyConnector` property is set on the token response, we retrieve the `masterKey`
+    from state and use it to decrypt and store the `userKey`.
+- **SSO strategy**
+  - Trusted Device Encryption presents us with multiple ways to obtain the User Key, and these are
+    handled here, in the SSO login strategy.
+  - These include:
+    - Using the device key if the user is on a Trusted Device
+    - Using an approved Admin Approval request to receive the key
+    - Using the master key if the user has a master password
+
+#### User Private Key (`privateKey`)
+
+If a `PrivateKey` is provided in the response, it is stored in `privateKey.encrypted` in the user's
+account state.
+
+## Appendix: `UserDecryptionOptions` Details
+
+The token response will contain an instance of `UserDecryptionOptions`. This object contains the
+following properties, which describe how a user will be able to decrypt their vault.
+
+- `HasMasterPassword` - Will be `true` if the user has a master password in the database.
+- `TrustedDeviceOption` - Contains the properties defining whether a user can decrypt their
+  symmetric key using Trusted Device Encryption.
+  - `HasAdminApproval` - Will be `true` if a user can use Admin Approval to request approval for a
+    new Trusted Device.
+  - `HasLoginApprovingDevice` - Will be `true` if a user has any eligible ("known") devices that can
```

Comment:
Should we be using "known" device here?  

In the standard login process, we call the `getKnownDevice` endpoint to determine if the user has logged in the device before (determines if it is "known").

In the TDE login with device flow, this check means the user has mobile or desktop devices which are capable of being configured to approve a login with device request (no guarantee they are setup for it or not though). Nothing here regarding the "known" status of the device or if they've logged in before or not.

Reply:
Absolutely right.  I was conflating the "known" check on _requesting_ Login with Device for authentication with the determination of eligible devices for _approving_ a request.  I've removed this.